roles/infrastructure/tasks/setup_aws_network.yml

```diff
@@ -40,29 +40,27 @@
   ansible.builtin.set_fact:
     infra__aws_igw_id: "{{ __aws_igw.gateway_id }}"
 
-- name: Create AWS VPC Subnets
+- name: Create AWS VPC Public Subnets
   amazon.aws.ec2_vpc_subnet:
     region: "{{ infra__region }}"
     vpc_id: "{{ infra__aws_vpc_id }}"
-    cidr: "{{ __aws_subnet_create_item.cidr }}"
+    cidr: "{{ __aws_public_subnet_item.cidr }}"
     state: present
-    tags: "{{ infra__tags | combine(__aws_subnet_create_item.tags, recursive=True) }}"
+    tags: "{{ infra__tags | combine(__aws_public_subnet_item.tags, recursive = true) }}"
     map_public: yes
     az: "{{ __aws_az_info.availability_zones[__aws_subnet_index % infra__aws_vpc_az_count | int].zone_name }}"
   loop_control:
-    loop_var: __aws_subnet_create_item
+    loop_var: __aws_public_subnet_item
     index_var: __aws_subnet_index
-    label: __aws_subnet_create_item.name
-  loop: "{{ infra__vpc_public_subnets_info | union(infra__vpc_private_subnets_info) }}"
-  register: __aws_subnets
+  loop: "{{ infra__vpc_public_subnets_info }}"
+  register: __aws_public_subnets
 
-- name: Set fact for AWS Subnet IDs
+- name: Set fact for AWS Public Subnet IDs
   ansible.builtin.set_fact:
-    infra__aws_subnet_ids: "{{ infra__aws_subnet_ids | default([]) | union([__aws_subnet_item.subnet.id | default('')]) }}"
+    infra__aws_public_subnet_ids: "{{ infra__aws_public_subnet_ids | default([]) | union([__aws_subnet_item.subnet.id | default('')]) }}"
   loop_control:
     loop_var: __aws_subnet_item
-    label: "{{ __aws_subnet_item.subnet.id }}"
```

Comment:
Include a label -- produces better output within Ansible

Reply:
I see. Thanks for tip!